metabase/metabase
# Action Creator 5: Use Action Form Settings In Dashboard Modal Forms

## Description

We want to get our form customization settings flowing through to actual usable forms on dashboards asap.

This PR creates a common code path for form previews in the action editor and form rendering in dashboards - mainly via the `getParameterFieldProps` function which builds actual form props from form settings, and all form fields are rendered via the various `FormWidgets` used by our common Form component.

## Screenshots

Action Creator:
![Screen Shot 2022-09-09 at 3 43 48 PM](https://user-images.githubusercontent.com/30528226/189449993-43929f2d-e09b-4a31-b1a4-56be28cf89a7.png)

Form on dashboard:
![Screen Shot 2022-09-09 at 3 43 05 PM](https://user-images.githubusercontent.com/30528226/189449992-a39ceabb-e8eb-4dae-8d30-2b958909b716.png)

## Testing Steps
- create a new action with multiple template tags - define different input types for each (except currently the select and radio options don't work because we haven't built the UI to define the options yet)
- see that you get a preview of the selected input types
- save the action
- add an action button to an app page, and connect the click action of that button to your newly created action
- see that clicking that action displays the fields you defined and the input types you defined
